core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/MultiregionalBucketMigrationActor.scala

```diff
@@ -0,0 +1,857 @@
+                                       getSrcBucket: (GcsBucketName, GcsBucketName) => GcsBucketName,
+                                       getDstBucket: (GcsBucketName, GcsBucketName) => GcsBucketName,
+                                       getColumn: DataAccess => ColumnName[Option[Timestamp]]
+  ): MigrateAction[Unit] =
+    withMigration(readMigrations) { (migration, workspace) =>
+      for {
+        (srcBucket, dstBucket) <- getSrcAndDstBuckets(migration, workspace, getSrcBucket, getDstBucket)
+
+        (googleProject, storageService, transferService) <- asks { d =>
+          (d.googleProjectToBill, d.storageService, d.storageTransferService)
+        }
+
+        _ <- liftIO {
+          for {
+            serviceAccount <- transferService.getStsServiceAccount(googleProject)
+            serviceAccountList = NonEmptyList.one(Identity.serviceAccount(serviceAccount.email.value))
+            // STS requires the following to read from the origin bucket and delete objects after
+            // transfer
+            _ <- storageService
+              .setIamPolicy(
+                srcBucket,
+                Map(StorageRole.LegacyBucketWriter -> serviceAccountList,
+                    StorageRole.ObjectViewer -> serviceAccountList
+                ),
+                bucketSourceOptions =
+                  if (migration.requesterPaysEnabled) List(BucketSourceOption.userProject(googleProject.value))
+                  else List.empty
+              )
+              .compile
+              .drain
+
+            // STS requires the following to write to the destination bucket.
+            // Note destination bucket will not have requester pays enabled until end of migration.
+            _ <- storageService
+              .setIamPolicy(
+                dstBucket,
+                Map(StorageRole.LegacyBucketWriter -> serviceAccountList,
+                    StorageRole.ObjectCreator -> serviceAccountList
+                )
+              )
+              .compile
+              .drain
+
+          } yield ()
+        }
+
+        now <- nowTimestamp
+        _ <- inTransaction { dataAccess =>
+          dataAccess.multiregionalBucketMigrationQuery.update(migration.id, getColumn(dataAccess), now.some)
+        }
+
+        _ <- getLogger[MigrateAction].info(
+          stringify(
+            "migrationId" -> migration.id,
+            "workspace" -> workspace.toWorkspaceName,
+            "srcBucket" -> srcBucket,
+            "dstBucket" -> dstBucket,
+            "iamConfigured" -> now
+          )
+        )
+      } yield ()
+    }
+
+  final def issueBucketTransferJob(readMigrations: DataAccess => ReadAction[Vector[MultiregionalBucketMigration]],
+                                   getSrcBucket: (GcsBucketName, GcsBucketName) => GcsBucketName,
+                                   getDstBucket: (GcsBucketName, GcsBucketName) => GcsBucketName,
+                                   getColumn: DataAccess => ColumnName[Option[Timestamp]]
```

Comment:
Nitpick (as I know this was copied from the existing PPW stuff): Took me a bit to grok what this was doing with the call on L615, might be better named `getTimestampColumnToUpdate` or something equally explicit.